HW2/P2/parallel_vector.pyx

```diff
@@ -80,12 +80,45 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    # data movement.  Be sure to avoid deadlock, and double-locking.
    ##########
    with nogil:
-       for r in range(repeat):
+       for r in prange(repeat, num_threads=4):
            for idx in range(src.shape[0]):
+               src_index_value = src[idx]
+               dest_index_value = dest[idx]
+
```

Comment:
what if src[idx] changes between now and acquiring the lock? You usually want to check the same condition again after acquiring the lock